mypy/checkexpr.py

```diff
@@ -1762,6 +1762,17 @@ def overload_arg_similarity(actual: Type, formal: Type) -> int:
     if isinstance(actual, UnionType):
         return max(overload_arg_similarity(item, formal)
                    for item in actual.items)
+    if isinstance(formal, TypeType):
```

Comment:
I wonder if this should go after the "if isinstance(formal, UnionType):" block below? Generally it seems unions are processed before non-union types. (The actual being a union is separate from the formal being a union.)